app/src/main/java/com/codefororlando/petadoption/recyclerview/HorizontalViewPagerIndicator.kt

```diff
@@ -0,0 +1,119 @@
+package com.codefororlando.petadoption.recyclerview
+
+import android.content.Context
+import android.support.v4.view.PagerAdapter
+import android.support.v4.view.ViewPager
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.animation.Animation
+import android.view.animation.ScaleAnimation
+import android.widget.LinearLayout
+import com.codefororlando.petadoption.R
+
+class HorizontalViewPagerIndicator : LinearLayout, ViewPager.OnPageChangeListener, ViewPager.OnAdapterChangeListener {
+
+    private var pageCount = 0
+    private var selectedIndex = 0
+    private var indicatorViews: MutableList<View> = mutableListOf()
+    private lateinit var scaleUpAnim: ScaleAnimation
+    private lateinit var scaleDownAnim: ScaleAnimation
+
+    constructor(context: Context) : super(context) {
```

Comment:
You don't need the repeating constructors if you use JvmOverloads.

```kotlin
class HorizontalViewPagerIndicator @JvmOverloads constructor(
   context: Context,
   attrs: AttributeSet? = null,
   defStyleAttr: Int = 0
) : LinearLayout, ViewPager.OnPageChangeListener, ViewPager.OnAdapterChangeListener {
```

Then remove the unnecessary `init()` function and use `init{}` then change your `lateinit` vars to `val`s

Reply:
Thanks for the tip.  Updated!